src/test/scala/zio/s3/S3SettingsTest.scala

```diff
@@ -2,26 +2,74 @@ package zio.s3
 
 import software.amazon.awssdk.regions.Region
 import zio.test.Assertion._
-import zio.test._
+import zio.test.{ testM, _ }
 
 object S3SettingsTest extends DefaultRunnableSpec {
 
   def spec =
-    suite("settings")(
-      testM("invalid region") {
-        for {
-          failure <- S3Settings
-                       .from(Region.of("invalid"), S3Credentials("key", "secret"))
-                       .foldCause(_.failureOption.map(_.message).mkString, _ => "")
-        } yield assert(failure)(equalTo("Invalid aws region provided : invalid"))
-      },
-      testM("valid region") {
-        for {
-          success <- S3Settings.from(Region.US_EAST_2, S3Credentials("key", "secret"))
-        } yield assert(success.s3Region.region -> success.credentials)(
-          equalTo(Region.US_EAST_2             -> S3Credentials("key", "secret"))
-        )
-      }
+    suite("Settings")(
+      suite("Regions")(
+        testM("invalid region") {
+          for {
+            failure <- S3Settings
+                         .from(Region.of("invalid"), S3Credentials("key", "secret"))
+                         .foldCause(_.failureOption.map(_.message).mkString, _ => "")
+          } yield assert(failure)(equalTo("Invalid aws region provided : invalid"))
+        },
+        testM("valid region") {
+          for {
+            success <- S3Settings.from(Region.US_EAST_2, S3Credentials("key", "secret"))
+          } yield assert(success.s3Region.region -> success.credentials)(
+            equalTo(Region.US_EAST_2             -> S3Credentials("key", "secret"))
+          )
+        }
+      ),
+      suite("credentials")(
+        testM("no cred in system properties") {
+          for {
+            failure <- S3Credentials.fromSystem.flip.map(_.message)
+          } yield assert(failure)(isNonEmptyString)
+        },
+        testM("cred in system properties") {
+          System.setProperty("aws.accessKeyId", "k1")
+          System.setProperty("aws.secretAccessKey", "s1")
+          for {
+            cred <- S3Credentials.fromSystem
+            _     = System.clearProperty("aws.accessKeyId")
+            _     = System.clearProperty("aws.secretAccessKey")
+          } yield assert(cred)(equalTo(S3Credentials("k1", "s1")))
+        },
```

Comment:
Not sure if it's critical for this test suite to clean up after itself, but I think it will keep sys props set if `S3Credentials.fromSystem` fails, right?

Reply:
[around_](https://github.com/zio/zio/blob/master/test/shared/src/main/scala/zio/test/TestAspect.scala#L126) is better fit for this.

Reply:
thank you or the trick